### PR TITLE
Change wide swing sprites to be that of the weapon used

### PR DESCRIPTION
**Pull request by notquitehadouken**
## About the PR

Changes the slash/thrust effect to have the same texture as the weapon used to create it.
also makes it swing right by default because the weapons that have blades all face down

## Why / Balance

Created as a response to PR #20754 (now closed), which was made into a draft due to needing to be implemented in code instead of creating a new prototype for the wide swing on literally every weapon capable of wide swinging
Looks cool.

## Technical details

Adds an EntityUid for the weapon used to all the event and function calls
Changes the sprite of the slash/thrust effect at runtime via the SpriteComponent
Adds two bools in the MeleeWeaponComponent, one says if the texture should be replaced (defaults to yes) and the other if the swing should be left to right instead of right to left, and an angle that says what rotation the sprite should be.
Allows the rotation of the slash/thrust sprite to be changed.
Accidentally cleans up AbsorbentSystem a little

## Media

https://github.com/space-wizards/space-station-14/assets/45953835/4a290dac-1f49-4c08-bc3c-4d81f9441955

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: The grey stick that shows up when you wide swing a weapon now looks like the weapon used.

### CODE REVIEW

**Comment by CatMagic1** [comment]
I wanna see da media

**Comment by potato1234x** [comment]
where video

**Comment by notquitehadouken** [comment]
gimme a bit i still have crap to fix

**Comment by notquitehadouken** [comment]
i am very close to fixing it

**Comment by notquitehadouken** [comment]
will do more things tomorrow

**Comment by notquitehadouken** [comment]
ALLLLLLLLLLMOST t h e r e

**Comment by notquitehadouken** [comment]
this may take longer than expected so i'm going to shut up until i finish

**Comment by notquitehadouken** [comment]
done, now will add video

**Comment by notquitehadouken** [comment]
NOW i am done

**Comment by notquitehadouken** [comment]
feel free to tell me if you find a melee weapon i missed (for some reason search all'ing "MeleeWeapon" doesn't give me all the results)

**Comment by CatMagic1** [comment]
Cool but.

Is it just me or are the sprites lookin swiggity swonkulous?

**Comment by notquitehadouken** [comment]
> Cool but.
> 
> Is it just me or are the sprites lookin swiggity swonkulous?

low resolution viewport

**Comment by notquitehadouken** [comment]
i tried to use high res but it dropped the fps to like 4

**Comment by CatMagic1** [comment]
Nice

**Comment by TheShuEd** [comment]
i love u!
this is perfect

**Comment by CatMagic1** [comment]
Armblade?

**Comment by notquitehadouken** [comment]
whoopy dasy

**Comment by CatMagic1** [comment]
The desword looks weird like this but I have no idea how it would animate swinging in a normal way

https://youtu.be/CvvAftMZYKM

File: Content.Client/Weapons/Melee/MeleeWeaponSystem.Effects.cs [hunk 1]
```diff
@@ -41,6 +42,19 @@ public override void DoLunge(EntityUid user, Angle angle, Vector2 localPos, stri
             return;
         }
 
+        var spriteRotation = Angle.Zero;
+        if (arcComponent.Animation != WeaponArcAnimation.None
+            && TryComp<MeleeWeaponComponent>(weapon, out var meleeWeaponComponent))
+        {
+            if (meleeWeaponComponent.UseWeaponAsAttackAnimation
+                && TryComp<SpriteComponent>(weapon, out var weaponSpriteComponent))
+                sprite.CopyFrom(weaponSpriteComponent);
```
Comment (metalgearsloth) [inline Content.Client/Weapons/Melee/MeleeWeaponSystem.Effects.cs:51]: Please use the other versions of TryComp

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml [hunk 2]
```diff
@@ -2001,6 +2008,7 @@
       Dead:
         Base: narsian_dead
   - type: MeleeWeapon
+    useWeaponAsAttackAnimation: false
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2011]: Setting this for every single mob is very tedious so please find an alternative.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 2 above)
Comment (deltanedas) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml:2011]: isnt there a base melee mob

**Review by metalgearsloth — APPROVED** [review]

File: Content.Client/Weapons/Melee/MeleeWeaponSystem.Effects.cs (hunk 1 above)
Comment (notquitehadouken) [reply Content.Client/Weapons/Melee/MeleeWeaponSystem.Effects.cs]: did

File: Resources/Prototypes/Entities/Mobs/NPCs/animals.yml (hunk 2 above)
Comment (notquitehadouken) [reply Resources/Prototypes/Entities/Mobs/NPCs/animals.yml]: removed entirely

**Comment by notquitehadouken** [comment]
uh oh it broke everything time to release a hotfix

**Pull request merged**